fs-cache/src/cache.rs

```diff
@@ -0,0 +1,668 @@
+    fn get_file_size(&self, key: &K) -> Result<usize> {
+        Ok(fs::metadata(self.path.join(key.to_string()))?.len() as usize)
+    }
+
+    /// Adds or updates a value in the memory cache, evicting old entries if needed.
+    /// Returns error if value is larger than maximum memory limit.
+    fn update_memory_cache(&mut self, key: &K, value: &V) -> Result<()> {
+        log::debug!("cache/{}: caching in memory for key {}", self.label, key);
+        let size = self.get_file_size(key)?;
+
+        // If single value is larger than total limit, just skip memory caching
+        if size > self.max_memory_bytes {
+            log::error!(
+                "cache/{}: value size {} exceeds limit {}",
+                self.label,
+                size,
+                self.max_memory_bytes
+            );
+            return Ok(());
+        }
+
+        // Remove oldest entries until we have space for new value
+        while self.current_memory_bytes + size > self.max_memory_bytes {
+            let (_, old_entry) = self
+                .memory_cache
+                .pop_lru()
+                .expect("Cache should have entries to evict");
+            debug_assert!(
+                self.current_memory_bytes >= old_entry.size,
+                "Memory tracking inconsistency detected"
+            );
+            self.current_memory_bytes = self
+                .current_memory_bytes
+                .saturating_sub(old_entry.size);
+        }
+
+        // Add new value and update size
+        self.memory_cache.put(
+            key.clone(),
+            CacheEntry {
+                value: value.clone(),
+                size,
+            },
+        );
+        self.current_memory_bytes += size;
+
+        log::debug!(
+            "cache/{}: added {} bytes, total {}/{}",
+            self.label,
+            size,
+            self.current_memory_bytes,
+            self.max_memory_bytes
+        );
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use rstest::{fixture, rstest};
+    use std::{
+        fs::File,
+        io::Write,
+        time::{Duration, SystemTime},
+    };
+    use tempdir::TempDir;
+
+    // Helper struct that implements required traits
+    #[derive(Clone, Debug, PartialEq)]
+    struct TestValue(Vec<u8>);
+
+    impl AsRef<[u8]> for TestValue {
+        fn as_ref(&self) -> &[u8] {
+            &self.0
+        }
+    }
+
+    impl From<Vec<u8>> for TestValue {
+        fn from(bytes: Vec<u8>) -> Self {
+            TestValue(bytes)
+        }
+    }
+
+    #[fixture]
+    fn temp_dir() -> TempDir {
+        TempDir::new("tmp").expect("Failed to create temporary directory")
+    }
+
+    #[fixture]
+    fn cache(temp_dir: TempDir) -> Cache<String, TestValue> {
+        Cache::new(
+            "test".to_string(),
+            temp_dir.path(),
+            1024 * 1024, // 1MB
+            false,
+        )
+        .expect("Failed to create cache")
+    }
```

Comment:
I feel like using `fixtures` here might be a bit overkill, but maybe you have plans to expand on this in the future. Personally, I think we could just create the `temp_dir` and `cache` instances directly in each test function, or use a helper function, which would make the code more readable.

Reply:
Done!